android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.remotesettings
+
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mozilla.appservices.remotesettings.RemoteSettings
+import mozilla.appservices.remotesettings.RemoteSettingsConfig
+import mozilla.appservices.remotesettings.RemoteSettingsException
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.BufferedReader
+import java.io.File
+import java.io.FileReader
+import java.io.FileWriter
+import java.io.IOException
+
+/**
+ * Helper class to download collections from remote settings in app services.
+ *
+ * @property storageRootDirectory The top level app-local storage directory.
+ * @property serverUrl The optional url for the settings server. If not specified, the standard server will be used.
+ * @property bucketName The optional name of the bucket containing the collection on the server.
+ * If not specified, the standard bucket will be used.
+ * @property collectionName The name of the collection for the settings server.
+ */
+class RemoteSettingsClient(
+    private val storageRootDirectory: File,
+    private val serverUrl: String? = "https://firefox.settings.services.mozilla.com",
+    var bucketName: String? = "main",
+    var collectionName: String = "",
+) {
+
+    private val config = RemoteSettingsConfig(
+        serverUrl = serverUrl,
+        bucketName = bucketName,
+        collectionName = collectionName,
+    )
+
+    /**
+     * Fetches all records for a collection that can be found in the local storage.
+     **/
+    suspend fun fetch(): JSONObject? = withContext(Dispatchers.IO) {
+        val fileName = "list.json"
```

Comment:
We should investigate a less opinionated file caching mechanisms for these records. I think we will at least want a way for consumers to be able to specify the name of the file they are saving a collection or record to. Happy to discuss design in more depth when I get back next week 🙂